sdc_client/client.py

```diff
@@ -312,17 +312,15 @@ def get_jmx(streamsets: IStreamSets, query: str) -> dict:
 
 
 def get_jmxes_async(queries: List[Tuple[IStreamSets, str]]) -> List[Dict]:
-    async def execute_requests(queries: List[Tuple[IStreamSets, str]]):
-        tasks = [asyncio.create_task(
-            _AsyncStreamSetsApiClient(streamset_).get_jmx(query)
-        ) for streamset_, query in queries]
-        await asyncio.wait(tasks)
+    async def execute_requests(queries_: List[Tuple[IStreamSets, str]]):
+        streamsets_ = set([streamset_ for streamset_, _ in queries_])
+        async with _AsyncClientsManager(streamsets_) as manager:
+            tasks = [asyncio.create_task(
+                manager.clients[streamset_].get_jmx(query)
+            ) for streamset_, query in queries_]
+            await asyncio.wait(tasks)
         return [task.result() for task in tasks]
-    loop = asyncio.new_event_loop()
-    jmxes = loop.run_until_complete(execute_requests(queries))
-    loop.close()
-    return jmxes
-
+    return asyncio.run(execute_requests(queries))
```

Comment:
As a suggestion, it could be simplified without inner function declaration and `asyncio.gather` usage, like so:

```python
def get_jmxes_async(queries: List[Tuple[IStreamSets, str]]):
    streamsets_ = list({streamset_ for streamset_, _ in queries})
    async with _AsyncClientsManager(streamsets_) as manager:
        return await asyncio.gather(
            *[asyncio.create_task(manager.clients[streamset_].get_jmx(query)) for streamset_, query in queries],
            return_exceptions=True
        )
...
jmxes = asyncio.run(get_jmxes_async(get_jmxes_async))
```